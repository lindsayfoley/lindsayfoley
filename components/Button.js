import React from "react";
import "../public/styles/_button.scss";

function Button({ link, cta }) {
  return (
    <a href={link}>
      <button>{cta}</button>
    </a>
  );
}

export default Button;
