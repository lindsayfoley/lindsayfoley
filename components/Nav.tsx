import React from "react";
import AnchorButton, { IAnchorButtonProps } from "./AnchorButton";
import "../public/styles/_anchorButton.scss";

interface INavProps {
  buttons: IAnchorButtonProps[];
}

const Nav: React.FunctionComponent<INavProps> = ({ buttons }) =>
  <nav>
    {buttons.map(button => <AnchorButton {...button} key={button[0].cta} />)}
  </nav>;

export default Nav;