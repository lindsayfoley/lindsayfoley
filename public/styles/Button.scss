@import "_base.scss";

button {
  width: 250px;
  @include primary-font;
  font-size: 18px;
  font-weight: bold;
  line-height: 35px;
  background: inherit;
  color: $light-color;
  border: $default-border;
  transition: all 0.3s ease 0s;

  &:hover {
    background: $light-color;
    color: $primary-colour;
  }
}