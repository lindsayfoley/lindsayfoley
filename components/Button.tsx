import React from "react";
import Link from "next/link";
import "../public/styles/Button.scss";

export interface IButtonProps {
  link: string;
  cta: string;
}

const Button: React.FunctionComponent<IButtonProps> = ({ link, cta }) =>
  <Link href={link}>
    <a className="button" itemProp="url">
      {cta}
    </a>
  </Link>;

export default Button;