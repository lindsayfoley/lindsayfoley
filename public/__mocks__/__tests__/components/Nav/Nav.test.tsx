import React from "react";
import { mount } from 'enzyme';
import Nav, { INavProps } from '../../../../../components/Nav';
import Button, { IButtonProps } from '../../../../../components/Button';

const mockButtonOne: IButtonProps = {
    link: "portfolio",
    cta: "View my work"
}

const mockButtonTwo: IButtonProps = {
    link: "/",
    cta: "back to home"
}

const defaultProps: INavProps = {
    buttonArr: [mockButtonOne, mockButtonTwo]
}

describe('Nav', () => {

    it('renders correctly', () => {
        const wrapper = mount(<Nav {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a nav with one button', () => {
        const wrapper = mount(<Nav buttonArr={[mockButtonTwo]} />);
        expect(wrapper.find(Button).length).toBe(1);
        expect(wrapper.find(Button).first().text()).toEqual(mockButtonTwo.cta);
    });

    it('renders a nav with multiple buttons', () => {
        const wrapper = mount(<Nav {...defaultProps} />);
        expect(wrapper.find(Button).length).toBe(2);
        expect(wrapper.find(Button).first().text()).toEqual(mockButtonOne.cta);
        expect(wrapper.find(Button).last().text()).toEqual(mockButtonTwo.cta);
    });
});