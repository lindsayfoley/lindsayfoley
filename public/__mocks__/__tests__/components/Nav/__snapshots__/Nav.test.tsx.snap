// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<Nav
  buttonArr={
    Array [
      Object {
        "cta": "View my work",
        "link": "portfolio",
      },
      Object {
        "cta": "back to home",
        "link": "/",
      },
    ]
  }
>
  <nav>
    <Button
      cta="View my work"
      key="0"
      link="portfolio"
    >
      <Link
        href="portfolio"
      >
        <button
          itemProp="url"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View my work
        </button>
      </Link>
    </Button>
    <Button
      cta="back to home"
      key="1"
      link="/"
    >
      <Link
        href="/"
      >
        <button
          itemProp="url"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          back to home
        </button>
      </Link>
    </Button>
  </nav>
</Nav>
`;
